src/main/java/com/vinyl/service/ValidatorFactory.java

```diff
@@ -0,0 +1,20 @@
+package com.vinyl.service;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class ValidatorFactory {
+
+	public Validator getValidator(String validatorType) {
```

Comment:
String type is too lax, you have to return null because you use it, a better approach would be to use an Enum type, this way the method would only receive valid values.
-----
However, I don't see the benefit of using the Enum or any other param type in this case. I think it would be better that you create 2 methods in this class, getUserNameValidator and getUserEmailValidator

Reply:
fixed